torch/utils/checkpoint.py

```diff
@@ -323,7 +323,17 @@ def _checkpoint_without_reentrant(function, preserve_rng_state=True, *args):
             the RNG state during each checkpoint.
         *args: Arguments to pass in to the given ``function``.
     """
-    had_autocast_in_fwd = torch.is_autocast_enabled()
+    # Accommodates the (remote) possibility that autocast is enabled for cpu AND gpu.
```

Comment:
nit: Should we refactor this commonality into something like `_get_autocast_kwargs()` that returns these two `dict`s to avoid the duplication with reentrant AC?